src/app/GitUI/UserControls/RevisionGrid/QuickSearchProvider.cs

```diff
@@ -56,37 +55,47 @@ public void OnKeyPress(KeyPressEventArgs e)
 
             curIndex = curIndex >= 0 ? curIndex : 0;
 
-            if (e.KeyChar == 8 && _quickSearchString.Length > 1)
+            if (e.KeyChar == (char)Keys.Back && _quickSearchString.Length > 1)
             {
                 // backspace
-                RestartQuickSearchTimer();
-
-                _quickSearchString = _quickSearchString[..^1];
-
-                FindNextMatch(curIndex, _quickSearchString, false);
-                _lastQuickSearchString = _quickSearchString;
-
-                e.Handled = true;
-                ShowQuickSearchString();
+                UpdateQuickSearchString(_quickSearchString[..^1]);
+            }
+            else if (e.KeyChar == 22 && Control.ModifierKeys == Keys.Control && Clipboard.ContainsText())
+            {
+                // paste
+                string text = Clipboard.GetText();
+                UpdateQuickSearchString(string.Concat(_quickSearchString, text));
             }
             else if (!char.IsControl(e.KeyChar))
+            {
+                // The code below is meant to fix the weird key values when pressing keys e.g. ".".
+                UpdateQuickSearchString(string.Concat(_quickSearchString, char.ToLower(e.KeyChar)));
+            }
+            else
+            {
+                HideQuickSearchString();
+                e.Handled = false;
+            }
+
+            void UpdateQuickSearchString(string newValue)
```

Comment:
We prefer to explicitly indicate where the function ends if there are local functions.
```suggestion
            return;

            void UpdateQuickSearchString(string newValue)
```